test/e2e/ipfs-companion.test.js

```diff
@@ -0,0 +1,176 @@
+    }
+  }
+  console.warn('No extension URL found')
+}
+
+async function updateExtensionSettings (browser, url, id, value) {
+  console.info('Updating extension setting')
+  console.info(`Going to: ${url}/dist/options/options.html`)
+  await browser.get(`${url}/dist/options/options.html`)
+  console.info(`Looking for an element: ${id}`)
+  let element = browser.findElement(By.id(id))
+  console.info(`Setting new value to: ${value}`)
+  await element.sendKeys('')
+  await delay(1000) // waiting for focus to be acquired
+  await element.clear()
+  await delay(1000) // waiting for input to be cleared
+  await element.sendKeys(value)
+  await element.sendKeys(Key.TAB)
+  await delay(5000) // waiting for the setting to be applied
+  console.info('Checking if the update worked')
+  element = browser.findElement(By.id(id))
+  const v = await element.getAttribute('value')
+  console.info('The setting update is complete')
+  return v
+}
+
+async function getNumberOfConnectedPeers (browser, url) {
+  console.info('Checking the number of connected peers')
+  console.info(`Going to: ${url}/dist/landing-pages/welcome/index.html`)
+  await browser.get(`${url}/dist/landing-pages/welcome/index.html`)
+  await delay(5000) // waiting for the connection number to appear
+  if (process.env.TEST_DEBUG === 'true') {
+    const html = await browser.getPageSource()
+    console.debug(html)
+  }
+  console.info('Looking for an element with text: \'Your node is connected to ...\'')
+  const p = browser.findElement(By.xpath("//p[text()='Your node is connected to ']"))
+  const span = p.findElement(By.css('span'))
+  const peers = await span.getText()
+  console.info(`There are ${peers} connected peers`)
+  return parseInt(peers)
+}
+
+async function runTest (browserName) {
+  const extension = getExtension(browserName)
+
+  console.info(`Checking if ${extension} exists`)
+  expect(fs.existsSync(extension)).to.be.true // eslint-disable-line no-unused-expressions
+
+  let browser
+  if (browserName === 'chromium') {
+    browser = await openChromium(extension)
+  } else if (browserName === 'firefox') {
+    browser = await openFirefox(extension)
+  } else {
+    fail(`unknown browser name: ${browserName}`)
+  }
+
+  try {
+    const url = await findExtensionUrl(browser)
+
+    expect(url).not.to.be.undefined // eslint-disable-line no-unused-expressions
+
+    const ipfsApiUrl = process.env.IPFS_API_URL || 'http://127.0.0.1:5001'
+    const updatedIpfsApiUrl = await updateExtensionSettings(browser, url, 'ipfsApiUrl', ipfsApiUrl)
+
+    expect(updatedIpfsApiUrl).to.equal(ipfsApiUrl)
+
+    const customGatewayUrl = process.env.CUSTOM_GATEWAY_URL || 'http://localhost:8080'
+    const updatedCustomGatewayUrl = await updateExtensionSettings(browser, url, 'customGatewayUrl', customGatewayUrl)
+
+    expect(updatedCustomGatewayUrl).to.equal(customGatewayUrl)
+
+    const peers = await getNumberOfConnectedPeers(browser, url)
+
+    expect(peers).not.to.equal(0)
```

Comment:
I feel like this will be flaky unless we have a guaranteed peer running in CI as well.